From a9421d55102fc84f77f7c21a2479fcd00652b896 Mon Sep 17 00:00:00 2001
From: Chen Qi <Qi.Chen@windriver.com>
Date: Mon, 25 Feb 2019 13:55:12 +0800
Subject: [PATCH 03/24] missing_type.h: add __compare_fn_t and comparison_fn_t

Make it work with musl where comparison_fn_t and __compare_fn_t
is not provided.

Upstream-Status: Inappropriate [musl specific]

Signed-off-by: Chen Qi <Qi.Chen@windriver.com>
---
 src/basic/missing_type.h | 9 +++++++++
 src/basic/util.h         | 1 +
 src/journal/catalog.c    | 1 +
 3 files changed, 11 insertions(+)

diff --git a/src/basic/missing_type.h b/src/basic/missing_type.h
index bf8a6ca..c487e65 100644
--- a/src/basic/missing_type.h
+++ b/src/basic/missing_type.h
@@ -10,3 +10,12 @@
 #if !HAVE_CHAR16_T
 #define char16_t uint16_t
 #endif
+
+#ifndef __GLIBC__
+typedef int (*comparison_fn_t)(const void *, const void *);
+#endif
+
+#ifndef __COMPAR_FN_T
+#define __COMPAR_FN_T
+typedef int (*__compar_fn_t)(const void *, const void *);
+#endif
diff --git a/src/basic/util.h b/src/basic/util.h
index 9f6a6ce..2c5dc32 100644
--- a/src/basic/util.h
+++ b/src/basic/util.h
@@ -26,6 +26,7 @@
 #include "format-util.h"
 #include "macro.h"
 #include "time-util.h"
+#include "missing.h"
 
 size_t page_size(void) _pure_;
 #define PAGE_ALIGN(l) ALIGN_TO((l), page_size())
diff --git a/src/journal/catalog.c b/src/journal/catalog.c
index 4062f12..034e00c 100644
--- a/src/journal/catalog.c
+++ b/src/journal/catalog.c
@@ -26,6 +26,7 @@
 #include "strv.h"
 #include "tmpfile-util.h"
 #include "util.h"
+#include "missing.h"
 
 const char * const catalog_file_dirs[] = {
         "/usr/local/lib/systemd/catalog/",
-- 
2.7.4

